subworkflows/local/illumina_mapping_depth.nf

```diff
@@ -9,19 +9,26 @@ include { SAMTOOLS_STATS                                  } from '../../modules/
 
 workflow ILLUMINA_MAPPING_DEPTH {
     take:
-    assemblies
+    assembly
     reads
 
     main:
     ch_versions = Channel.empty()
 
     // build index
-    BOWTIE2_ASSEMBLY_BUILD(assemblies)
+    BOWTIE2_ASSEMBLY_BUILD(assembly)
     ch_versions = ch_versions.mix(BOWTIE2_ASSEMBLY_BUILD.out.versions)
     ch_index = BOWTIE2_ASSEMBLY_BUILD.out.index
 
+    // match index to corresponding reads
+    ch_reads = reads.map{meta, reads -> [meta.id, meta, reads]}
+    ch_mapping = ch_index
+        .map{meta, index -> [meta.id, meta, index]}
+        .combine(ch_reads, by:0)
+        .map{id, index_meta, index, reads_meta, reads -> [index_meta, index, reads_meta, reads]}
```

Comment:
similar to the PR in the other repo, I don't totally understand the strategy you're using here (not a problem with the code, I don't understand the groovy enough), but this seems like the core piece of the PR so...

Reply:
without this the workflow would queue up a random set of reads to map to the assembly, whereas right now I need the reads that made up that assembly (the corresponding ones) to map to for downstream mapping stats and depth calculations. A later addition to this workflow is if you provide a parameter to do reciprocal mapping for a time-series for example then it will do reciprocal all-v-all, but for now I just want self-v-self for the same sample of the assembly and reads. I'm pretty sure from the reading and digging I did that the `.combine` and subsequent `.map` achieves this